frontend/js/src/brainzplayer/BrainzPlayerUI.tsx

```diff
@@ -149,6 +149,15 @@ function BrainzPlayerUI(props: React.PropsWithChildren<BrainzPlayerUIProps>) {
       }
     }
   }
+
+  function getURLFromID(trackID: string | undefined): string | undefined {
+    if (currentDataSourceName === "youtube") {
+      return `https://www.youtube.com/watch?v=${trackID}`;
+    }
+
+    return undefined;
+  }
+
```

Comment:
This isn't what we want to do. Currently, this completely overwrites any valid track url if the listening service is not youtube.
This is also not the right place to fix the issue with the trackUrl. You need to follow the prop up along the river to find why we are sending a videoID instead of a URL.

If you look in the YoutubePlayer component, you'll see we do the right thing here: https://github.com/metabrainz/listenbrainz-server/blob/frontend-redesign/frontend/js/src/brainzplayer/YoutubePlayer.tsx#L146
But we do the wrong thing here: https://github.com/metabrainz/listenbrainz-server/blob/frontend-redesign/frontend/js/src/brainzplayer/YoutubePlayer.tsx#L186

Ideally, this piece of code would be refactored in a static function (getURLFromVideoId)  so that it can be used for both, like we do with getURLFromListen
https://github.com/metabrainz/listenbrainz-server/blob/frontend-redesign/frontend/js/src/brainzplayer/YoutubePlayer.tsx#L100